docs/source/daml/intro/13_Interfaces.rst

```diff
@@ -0,0 +1,249 @@
+   whose contents are the result of applying the ``toTransferProposal`` method
+   to the current contract and the ``newOwner`` field of the choice argument.
+
+.. hint::
+  For a detailed explanation of the syntax used here, check out :ref:`daml-ref-interfaces`
+
+
+Interface instances
+-------------------
+
+On its own, an interface isn't very useful, since all contracts on the ledger
+must belong to some template type. In order to make the link between an
+interface and a template, we define an interface instance on either the
+template or the interface. In this example, we add
+``interface instance IAsset for Cash`` and
+``interface instance IAssetTransferProposal for CashTransferProposal``:
+
+.. literalinclude:: daml/daml-intro-13/daml/Cash.daml
+  :language: daml
+  :start-after: -- INTERFACE_INSTANCE_IASSET_FOR_CASH_BEGIN
+  :end-before: -- INTERFACE_INSTANCE_IASSET_FOR_CASH_END
+
+.. literalinclude:: daml/daml-intro-13/daml/Cash.daml
+  :language: daml
+  :start-after: -- INTERFACE_INSTANCE_IASSET_TF_FOR_CASH_TF_BEGIN
+  :end-before: -- INTERFACE_INSTANCE_IASSET_TF_FOR_CASH_TF_END
+
+The corresponding interface instances for ``NFT`` and ``NFTTransferProposal``
+are very similar so we omit them here.
+
+Inside the interface instances, we must implement every method defined for the
+corresponding interface, including the special ``view`` method.  Within each
+method implementation the variable ``this`` is in scope, corresponding to the
+implict current contract, which will have the type of the template (in this case
+``Cash`` / ``CashTransferProposal``), as well as each of the fields of the
+template type. For example, the ``view`` definition in ``interface instance
+IAsset for Cash`` mentions ``issuer`` and ``owner``, which refer to the issuer
+and owner of the current ``Cash`` contract, as well as ``this``, which refers to
+the entire ``Cash`` contract payload.
+
+The implementations given for each method must match the types given in the
+interface definition. In particular, we see that the ``view`` definition
+discussed above returns a ``VAsset``, corresponding to ``IAsset``'s
+``viewtype``. Similarly, ``setOwner`` returns an ``IAsset``, and
+``toTransferProposal`` returns an ``IAssetTransferProposal``. In these last two,
+we have used the function ``toInterface`` to convert values from a template type
+into an interface type. In ``setOwner``, ``toInterface`` is applied to a
+``Cash`` value (``this with owner = newOwner``), producing an ``IAsset`` value;
+in ``toTransferProposal``, it's applied to a ``CashTransferProposal`` value
+(``CashTransferProposal with {...}``), producing an ``IAssetTransferProposal``
+value.
+
+
+Using an interface
+------------------
+
+Now that we have some interfaces and templates with instances for them, we can
+reduce duplication in the code for different templates by instead going through
+the common interface.
+
+For instance, both ``Cash`` and ``NFT`` are ``Asset``s, which means that
```

Comment:
It looks like reStructuredText requires there to be a space after the `\`\`` to terminal a monospaced text section. Currently the monospaced text section goes past `Asset` because the "s" has no space separating it from the code section.

![image](https://user-images.githubusercontent.com/106664681/211784713-388b8247-d6e4-488d-9dec-e564ecf073f9.png)

Reply:
ah, thank you for spotting this!

I'll try with

```rst
``Asset``\s
```
per https://docutils.sourceforge.io/docs/ref/rst/restructuredtext.html#character-level-inline-markup-1; it's "not encouraged" but seems like the only option, and there are a few instances of `the ``foo``\s` in our `.rst`s already